lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -542,23 +575,39 @@ def _can_service_start(self) -> bool:
 
         return True
 
-    def _initialize_admin_user(self):
+    def _initialize_admin_user(self, pwd: Optional[str] = None):
         """Change default password of Admin user."""
-        hashed_pwd, pwd = generate_hashed_password()
-        self.secrets.put(Scope.APP, "admin_password", pwd)
+        is_update = pwd is not None
+        hashed_pwd, pwd = generate_hashed_password(pwd)
+
+        if is_update:
```

Comment:
Suggestion: perhaps this could be a separate function named `update_admin_user`? It's no longer being initialised in this case.

Reply:
I renamed it to `_put_admin_user`, as what we're essentially doing is in fact not initializing it by updating it in both cases, first case on disk where the initial stage user:pwd is `admin:admin` to `admin:generated_pwd` and the second case using the API to achieve the same